cdap-ui/app/cdap/components/PreviewData/Table.tsx

```diff
@@ -0,0 +1,140 @@
+/*
+ * Copyright © 2020 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+import React from 'react';
+import Table from '@material-ui/core/Table';
+import TableBody from '@material-ui/core/TableBody';
+import TableHead from '@material-ui/core/TableHead';
+import TableRow from '@material-ui/core/TableRow';
+import Paper from '@material-ui/core/Paper';
+import TableCell from '@material-ui/core/TableCell';
+import withStyles, { WithStyles } from '@material-ui/core/styles/withStyles';
+import isEmpty from 'lodash/isEmpty';
+import { PREVIEW_STATUS } from 'services/PreviewStatus';
+import Heading, { HeadingTypes } from 'components/Heading';
+import ThemeWrapper from 'components/ThemeWrapper';
+
+const CustomTableCell = withStyles((theme) => ({
+  head: {
+    backgroundColor: theme.palette.grey['300'],
+    color: theme.palette.common.white,
+    padding: 10,
+    fontSize: 14,
+  },
+  body: {
+    padding: 10,
+    fontSize: 14,
+  },
+}))(TableCell);
+
+export const messageTextStyle = {
+  fontSize: '1.3rem !important',
+  margin: '10px 0',
+};
+const styles = (theme) => ({
+  root: {
+    width: '100%',
+    display: 'inline-block',
+    height: 'auto',
+    marginTop: theme.spacing(1),
+  },
+  table: {
+    width: '100%',
+  },
+  row: {
+    height: 40,
+    '&:nth-of-type(odd)': {
+      backgroundColor: theme.palette.grey['600'],
+    },
+  },
+  messageText: messageTextStyle,
+});
+
+interface IDataTableProps extends WithStyles<typeof styles> {
+  headers: string[];
+  records: any[];
```

Comment:
Could this be represented in any specific type by any chance?

Reply:
That's a good question - as returned by the backend, the records are all objects but other than that the keys can be pretty much anything, and same for the values. So I don't know that having a type makes sense here.

Reply:
Do we do any parsing of the results from the backend?

Reply:
We parse the backend response to get the props (headers and records), but the records array hasn't been parsed further besides getting it out of the backend response. The prop records is an array of objects (or records) where each object corresponds to a row of a dataset. The keys are the field names (from the dataset schema), and the values are the values of the field for that row. So the keys depend on the dataset, and the values can be pretty much any type/value.